collector/factory.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package collector // import "go.opentelemetry.io/ebpf-profiler/collector"
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/xreceiver"
+
+	"go.opentelemetry.io/ebpf-profiler/collector/internal"
+	"go.opentelemetry.io/ebpf-profiler/internal/controller"
+)
+
+var (
+	typeStr = component.MustNewType("profiling")
+
+	errInvalidConfig = errors.New("invalid config")
+)
+
+// NewFactory creates a factory for the receiver.
+func NewFactory() receiver.Factory {
+	return xreceiver.NewFactory(
+		typeStr,
+		defaultConfig,
+		xreceiver.WithProfiles(createProfilesReceiver, component.StabilityLevelAlpha))
+}
+
+func createProfilesReceiver(
+	_ context.Context,
+	_ receiver.Settings, //nolint:gocritic // we must respect the collector API
+	baseCfg component.Config,
+	nextConsumer xconsumer.Profiles) (xreceiver.Profiles, error) {
+	cfg, ok := baseCfg.(*controller.Config)
+	if !ok {
+		return nil, errInvalidConfig
+	}
+
+	return internal.NewController(cfg, nextConsumer)
+}
+
+// todo: export default values (currently in main.go)
```

Comment:
not a blocker: Can you create an issue for this, if not done already?

Reply:
I've reviewed those values with the ones set in `cli_flags.go`, and added the only one we weren't setting.